sdk/identity/azure-identity/tests/test_cae.py

```diff
@@ -0,0 +1,139 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import base64
+import json
+import os
+import time
+
+from azure.core.pipeline.transport import HttpRequest
+from azure.core.exceptions import ResourceNotFoundError
+from azure.identity import (
+    AzureAuthorityHosts,
+    DeviceCodeCredential,
+    UsernamePasswordCredential,
+    InteractiveBrowserCredential,
+)
+from azure.identity._constants import DEVELOPER_SIGN_ON_CLIENT_ID
+from azure.mgmt.resource.subscriptions import SubscriptionClient
+from azure_devtools.scenario_tests import GeneralNameReplacer, RequestUrlNormalizer, patch_time_sleep_api
+from devtools_testutils import AzureTestCase
+import pytest
+import requests
+from six.moves.urllib_parse import urlparse
+
+from recording_processors import IdTokenProcessor, RecordingRedactor
+
+
+@pytest.mark.skip("these tests require support in azure-core")
+class CaeTestCase(AzureTestCase):
+    def __init__(self, *args, **kwargs):
+        scrubber = GeneralNameReplacer()
+        super(CaeTestCase, self).__init__(
+            *args,
+            recording_processors=[RecordingRedactor(), scrubber],
+            replay_processors=[RequestUrlNormalizer(), IdTokenProcessor()],
+            **kwargs
+        )
+        self.scrubber = scrubber
+        if self.is_live:
+            if "CAE_TENANT_ID" not in os.environ:
+                pytest.skip("Missing a tenant ID for CAE tests")
+            if "CAE_ARM_URL" not in os.environ:
+                pytest.skip("Missing an ARM URL for CAE tests")
+
+            self.cae_settings = {
+                "arm_scope": os.environ.get("CAE_ARM_SCOPE", "https://management.azure.com/.default"),
+                "arm_url": os.environ["CAE_ARM_URL"],
+                "authority": os.environ.get("CAE_AUTHORITY", AzureAuthorityHosts.AZURE_PUBLIC_CLOUD),
+                "graph_url": os.environ.get("CAE_GRAPH_URL", "https://graph.microsoft.com"),
+                "password": os.environ.get("CAE_PASSWORD"),
+                "tenant_id": os.environ["CAE_TENANT_ID"],
+                "username": os.environ.get("CAE_USERNAME"),
+            }
+            real = urlparse(self.cae_settings["arm_url"])
+            self.scrubber.register_name_pair(real.netloc, "management.azure.com")
+            self.scrubber.register_name_pair(self.cae_settings["tenant_id"], "tenant")
+            self.scrubber.register_name_pair(self.cae_settings["username"], "username")
```

Comment:
```suggestion
            self.scrubber.register_name_pair(self.cae_settings["username"], "username")
            self.scrubber.register_name_pair(self.cae_settings["password"], "password")
```
Would this be necessary as well, or are passwords already hidden in recordings?

Reply:
Passwords are in request bodies, which aren't recorded.